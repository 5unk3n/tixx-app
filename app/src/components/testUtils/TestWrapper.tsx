import { NavigationContainer } from '@react-navigation/native'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import React, { PropsWithChildren } from 'react'
import { PaperProvider } from 'react-native-paper'

import { theme } from '@/theme/defaultTheme'

const queryClient = new QueryClient({
	defaultOptions: {
		queries: {
			retry: false,
			staleTime: Infinity
		}
	}
})

export const TestWrapper = ({ children }: PropsWithChildren) => {
	return (
		<PaperProvider theme={theme.dark}>
			<QueryClientProvider client={queryClient}>
				<NavigationContainer>{children}</NavigationContainer>
			</QueryClientProvider>
		</PaperProvider>
	)
}
