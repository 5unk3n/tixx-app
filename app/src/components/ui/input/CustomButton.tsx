import React from 'react'
import { StyleSheet } from 'react-native'
import { Button, ButtonProps } from 'react-native-paper'
import { ThemeProp } from 'react-native-paper/lib/typescript/types'

import { useCustomTheme } from '@/hooks/useCustomTheme'
import { FontType } from '@/theme/defaultTheme'

export interface CustomButtonProps extends ButtonProps {
	size?: 'sm' | 'md' | 'lg'
	colorVariant?: 'primary' | 'secondary'
	labelVariant?: FontType
	flex?: boolean // 테일윈드로 flex 속성이 안 먹힘
}

export default function CustomButton({
	mode = 'outlined',
	size = 'lg',
	style,
	labelStyle,
	disabled,
	colorVariant = 'primary',
	labelVariant,
	flex = false,
	...props
}: CustomButtonProps) {
	const { colors, fonts } = useCustomTheme()

	const containerStyles = [
		styles.baseContainer,
		mode === 'outlined' && {
			backgroundColor: colors.grayscale[800]
		},
		size === 'md'
			? styles.mdContainer
			: size === 'sm'
				? styles.smContainer
				: styles.baseContainer,
		colorVariant === 'secondary' && {
			backgroundColor: colors.secondary,
			borderWidth: 0
		},
		disabled &&
			size === 'sm' && { backgroundColor: colors.point[900], borderWidth: 0 },
		disabled &&
			mode === 'outlined' && {
				backgroundColor: colors.grayscale[700]
			},
		flex && { flex: 1 },
		style
	]

	const labelStyles = [
		styles.baseLabel,
		size === 'lg'
			? [styles.lgLabel, fonts.body1Medium]
			: [styles.smLabel, fonts.body3Medium],
		colorVariant === 'secondary' && { color: colors.onSecondary },
		disabled &&
			size === 'sm' && {
				backgroundColor: colors.point[900],
				color: colors.grayscale[700]
			},
		labelVariant && fonts[labelVariant],
		labelStyle
	]

	const getThemes = (): ThemeProp => {
		switch (mode) {
			case 'contained':
				return {
					colors: { onPrimary: colors.grayscale[900] }
				}
			case 'outlined':
				return {
					colors: {
						primary: colors.grayscale[0],
						outline: colors.primary
					}
				}
			case 'text':
				return { colors: { primary: colors.onPrimary } }
			default:
				return {}
		}
	}

	return (
		<Button
			mode={mode}
			style={containerStyles}
			contentStyle={styles.baseContent}
			labelStyle={labelStyles}
			disabled={disabled}
			theme={getThemes()}
			{...props}
		/>
	)
}

const styles = StyleSheet.create({
	baseContainer: {
		height: 56,
		minWidth: 0,
		borderRadius: 8
	},
	mdContainer: {
		height: 48
	},
	smContainer: {
		height: 32,
		alignSelf: 'flex-start'
	},
	baseContent: {
		height: '100%'
	},
	baseLabel: {
		marginVertical: 0,
		marginHorizontal: 0
	},
	lgLabel: {
		marginHorizontal: 22
	},
	smLabel: {
		marginVertical: 8,
		marginHorizontal: 12
	}
})
