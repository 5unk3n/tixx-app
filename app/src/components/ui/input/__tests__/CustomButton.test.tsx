import { describe, it, expect, jest } from '@jest/globals'
import { render, screen, fireEvent } from '@testing-library/react-native'
import React from 'react'

import { TestWrapper } from '../../../testUtils/TestWrapper'
import CustomButton, { CustomButtonProps } from '../CustomButton'

describe('CustomButton', () => {
	it('should call onPress handler when clicked', () => {
		const onPressMock = jest.fn()
		render(
			<CustomButton onPress={onPressMock} accessibilityLabel="custom-btn">
				확인
			</CustomButton>,
			{ wrapper: TestWrapper }
		)

		const button = screen.getByLabelText('custom-btn')
		fireEvent.press(button)

		expect(onPressMock).toHaveBeenCalledTimes(1)
	})

	describe('should apply styles based on props', () => {
		// Define various combinations of props to test
		const testCases: Array<Partial<CustomButtonProps>> = [
			{ size: 'sm', mode: 'contained' },
			{ size: 'md', mode: 'outlined' },
			{ size: 'lg', mode: 'text' },
			{ colorVariant: 'primary', disabled: true },
			{ colorVariant: 'secondary', mode: 'contained' },
			{ flex: true },
			{ labelVariant: 'headline1Semibold' }
		]

		// Dynamically create a test for each combination
		testCases.forEach((props, i) => {
			it(`renders correctly for props combination ${i + 1}`, () => {
				const { toJSON } = render(
					<CustomButton {...props}>버튼</CustomButton>,
					{ wrapper: TestWrapper }
				)
				// Compare the rendered output with a stored snapshot
				expect(toJSON()).toMatchSnapshot()
			})
		})
	})
})
