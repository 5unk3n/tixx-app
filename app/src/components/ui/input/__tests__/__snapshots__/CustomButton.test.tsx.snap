// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should apply styles based on props renders correctly for props combination 1 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "#F2F862",
          "borderRadius": 8,
          "height": 32,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#F2F862",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 12,
                            "marginVertical": 8,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 16,
                          },
                        ],
                        false,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 2 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#1B1B1B",
          "borderRadius": 8,
          "height": 48,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#1B1B1B",
            "borderColor": "#F2F862",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 12,
                            "marginVertical": 8,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 16,
                          },
                        ],
                        false,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 3 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 22,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 18,
                          },
                        ],
                        false,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 4 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#2F2F30",
          "borderRadius": 8,
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#2F2F30",
            "borderColor": "#2F2F30",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#A0A2A2",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 22,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 18,
                          },
                        ],
                        false,
                        false,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 5 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#2F2F30",
          "borderRadius": 8,
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#2F2F30",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 22,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 18,
                          },
                        ],
                        {
                          "color": "#A0A2A2",
                        },
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 6 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#1B1B1B",
          "borderRadius": 8,
          "flex": 1,
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#1B1B1B",
            "borderColor": "#F2F862",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 22,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 18,
                          },
                        ],
                        false,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`CustomButton should apply styles based on props renders correctly for props combination 7 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#1B1B1B",
          "borderRadius": 8,
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#1B1B1B",
            "borderColor": "#F2F862",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "height": "100%",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#FFFFFF",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 0,
                          "marginVertical": 0,
                        },
                        [
                          {
                            "marginHorizontal": 22,
                          },
                          {
                            "fontFamily": "Pretendard-Medium",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": -0.02,
                            "lineHeight": 18,
                          },
                        ],
                        false,
                        undefined,
                        {
                          "fontFamily": "Pretendard-SemiBold",
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": -0.02,
                          "lineHeight": 28,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              버튼
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
